modules/openy_gc_personal_training/src/PersonalTrainingAccessControlHandler.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Drupal\openy_gc_personal_training;
+
+use Drupal\Core\Entity\EntityAccessControlHandler;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Access\AccessResult;
+
+/**
+ * Access controller for the Personal training entity.
+ *
+ * @see \Drupal\openy_gc_personal_training\Entity\PersonalTraining.
+ */
+class PersonalTrainingAccessControlHandler extends EntityAccessControlHandler {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function checkAccess(EntityInterface $entity, $operation, AccountInterface $account) {
+    switch ($operation) {
+      case 'view':
+        $active_provider = \Drupal::config('openy_gc_personal_training.settings')->get('active_provider');
```

Comment:
Can we add DI here?

Reply:
No, there will be fatal error in this case